webapi/src/main/scala/org/knora/webapi/util/StringFormatter.scala

```diff
@@ -232,12 +232,15 @@ class StringFormatter private(knoraApiHttpBaseUrl: String) {
     private val NCNamePattern: String =
     """[\p{L}_][\p{L}0-9_.-]*"""
 
-    // A regex sub-pattern for project ideas, which consist of at least 4 hexadecimal digits.
+    // A regex for matching a string containing only an ontology prefix label or a local entity name.
+    private val NCNameRegex: Regex = ("^" + NCNamePattern + "$").r
+
+    // A regex sub-pattern for project IDs, which consist of at least 4 hexadecimal digits.
     private val ProjectIDPattern: String =
         """\p{XDigit}{4,}"""
 
-    // A regex for matching a string containing only an ontology prefix label or a local entity name.
-    private val NCNameRegex: Regex = ("^" + NCNamePattern + "$").r
+    // A regex for matching a string containing the project ID.
+    private val ProjectIDRegex: Regex = (ProjectIDPattern).r
```

Comment:
This regex matches hex digits anywhere in a string:

```
scala> import scala.util.matching.Regex
import scala.util.matching.Regex

scala> val ProjectIDPattern: String = """\p{XDigit}{4,}"""
ProjectIDPattern: String = \p{XDigit}{4,}

scala> val ProjectIDRegex: Regex = (ProjectIDPattern).r
ProjectIDRegex: scala.util.matching.Regex = \p{XDigit}{4,}

scala> ProjectIDRegex.findFirstIn("****1234****")
res0: Option[String] = Some(1234)
```

I think you need this:

```
scala> val ProjectIDRegex: Regex = ("^" + ProjectIDPattern + "$").r
ProjectIDRegex: scala.util.matching.Regex = ^\p{XDigit}{4,}$

scala> ProjectIDRegex.findFirstIn("****1234****")
res1: Option[String] = None

scala> ProjectIDRegex.findFirstIn("1234")
res2: Option[String] = Some(1234)
```

Reply:
Thanks. I have also tightened it up (I think), to require 4 digits.